core/org.eclipse.birt.core/src/org/eclipse/birt/core/data/DateFormatISO8601.java

```diff
@@ -42,17 +43,39 @@
 	 */
 	public static Date parse( String source, TimeZone timeZone ) throws BirtException,
 			ParseException
+	{
+		DateFormat dateFormat = getDateFormat( source, timeZone );
+		Date resultDate = null;
+		try
+		{
+			if ( timeZone != null )
+			{
+				dateFormat.setTimeZone( timeZone );
+			}
+			resultDate = dateFormat.parse( source );
+			return resultDate;
+		}
+		catch ( ParseException e )
+		{
+			throw new CoreException( ResourceConstants.CONVERT_FAILS,
+					new Object[]{source.toString( ), "Date"} );
+		}
+	}
+
+	public static SimpleDateFormat getDateFormat( String source, TimeZone timeZone ) throws BirtException,
```

Comment:
Add comments and since.
If SimpleDataFormat is returned, please consider changing name to getSimpleDateFormat.